components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -222,41 +236,28 @@ let selectedPendingTransactionId: string = ''
       return this._accountsRegistry
     }
 
-    getSelectedAccountAddress = async () => {
-      if (!this._selectedAccountAddress) {
-        const { braveWalletService, keyringService } =
-          apiProxyFetcher()
-        const { coin: selectedCoin } =
-          await braveWalletService.getSelectedCoin()
-
-        if (selectedCoin === BraveWallet.CoinType.FIL) {
-          const { chainId }
-            = await braveWalletService.getChainIdForActiveOrigin(selectedCoin)
-          const { address } = await keyringService.getFilecoinSelectedAccount(
-            chainId
-          )
-          this._selectedAccountAddress = address
-        } else {
-          const { address } = await keyringService.getSelectedAccount(
-            selectedCoin
-          )
-          this._selectedAccountAddress = address
-        }
+    getSelectedAccount = async () => {
+      if (!this._selectedAccount) {
+        const allAccounts = await this.getAllAcountsInfo()
+        const info = allAccounts.accounts[allAccounts.selectedIndex];
+        this._selectedAccount = accountInfoToAccountInfoEntity(info)
       }
-      return this._selectedAccountAddress
+      return this._selectedAccount
     }
 
     clearWalletInfo = () => {
       this._walletInfo = undefined
+      this._allAccountsInfo = undefined
       this._accountsRegistry = undefined
+      this._selectedAccount = undefined
```

Comment:
Now that wallet info and accounts are separate, we should clear the caches for these separately